client/src/components/IframePreview.jsx

```diff
@@ -30,18 +30,20 @@ const IframePreview = forwardRef(({ slug, isLoading, device = 'iPhone X' }, ref)
     <iframe
       ref={iframeRef}
       src={`${import.meta.env.VITE_BACKEND_URL}/site/${slug}/`}
-      className="w-full h-full border-0"
+      className={`w-full h-full border-0  ${device === "iPhone X" && "pt-8 bg-black"}`}
```

Comment:
can add a hasNotch or needsSafeAreaAdjustment array which includes "iPhone X" and add a check here if current device is part of that array then add the padding top bg etc